arrays/arrays.c

```diff
@@ -57,13 +57,22 @@ Array *create_array (int capacity) {
 //  *****/
 void resize_array(Array *arr) {
 
-  // Create a new element storage with double capacity
+    // Create a new element storage with double capacity
+    char **doubleStorage = calloc(arr->capacity * 2, sizeof(char *));
 
-  // Copy elements into the new storage
-
-  // Free the old elements array (but NOT the strings they point to)
+    // Copy elements into the new storage
+    for (int i = 0; i < arr->count; ++i) {
+        doubleStorage[i] = arr->elements[i];
+    }
+    // Free the old elements array (but NOT the strings they point to)
```

Comment:
This for loop may be unnecessary as it frees the strings the old elements points to. Everything else in this function looks correct.